src/rulesets/Random/random.jl

```diff
@@ -19,11 +19,6 @@ end
 @non_differentiable rand(::Type{<:Real}, ::Integer...)
 @non_differentiable rand(::Integer...)
 
-# There are many different 1-3 arg methods, but not varargs
-@non_differentiable rand!(::Any)
-@non_differentiable rand!(::Any, ::Any)
-@non_differentiable rand!(::Any, ::Any, ::Any)
-
```

Comment:
Maybe keep less general versions such as `rand!(::AbstractArray)` and `rand!(::AbstractRNG, ::AbstractArray)`, similar to `randn!`?

Reply:
I've added back in a couple of methods.